ovs/extensions/healthcheck/arakoon/arakooncluster_health_check.py

```diff
@@ -282,179 +257,89 @@ def check_collapse(logger, arakoon_clusters=None, max_collapse_age=MAX_COLLAPSE_
-                                   .format(len(ver_result[3]), ', '.join(ver_result[3])),
-                                   'arakoon_unknown_exception')
-                else:
-                    logger.failure("Some Arakoon(s) have problems, please check this!", 'arakoon_integrity')
-            # check amount UNKNOWN_ERRORS arakoons
-            if len(ver_result[3]) > 0:
-                logger.failure("{0} Arakoon(s) seem(s) to have UNKNOWN ERRORS, please check the logs @"
-                               " '/var/log/ovs/arakoon.log' or"
-                               " '/var/log/upstart/ovs-arakoon-*.log': {1}".format(len(ver_result[3]),
-                                                                                   ', '.join(
-                                                                                       ver_result[3])),
-                               'arakoon_unknown_exception')
-            else:
-                logger.failure("Some Arakoon(s) have problems, please check this!", 'arakoon_integrity')
-
-            return working_arakoon_list, no_master_arakoon_list, down_arakoon_list, unkown_arakoon_list
-
+                try:
+                    # determine if there is a healthy cluster
+                    client = PyrakoonStore(str(cluster_name))
+                    client.nop()
+                    result_handler.success('Arakoon {0} responded successfully.'.format(cluster_name), test_name)
+                except ArakoonNotFound as ex:
+                    result_handler.failure('Arakoon {0} seems to be down. Got {1}'.format(cluster_name, str(ex)), test_name)
+                except (ArakoonNoMaster, ArakoonNoMasterResult) as ex:
+                    result_handler.failure('Arakoon {0} cannot find a master. Got {1}'.format(cluster_name, str(ex)), test_name)
+                except Exception as ex:
+                    result_handler.exception('Arakoon {0} could not process a nop. Got {1}'.format(cluster_name, str(ex)), test_name)
+
+    # @todo: separate cluster-wide-check
     @staticmethod
-    @expose_to_cli('arakoon', 'check-arakoons')
-    def check_arakoons(logger):
+    @expose_to_cli('arakoon', 'missing-node-test')
+    def check_arakoons(result_handler):
         """
         Verifies/validates the integrity of all available arakoons
 
-        :param logger: logging object
-        :type logger: ovs.log.healthcheck_logHandler.HCLogHandler
+        :param result_handler: logging object
+        :type result_handler: ovs.extensions.healthcheck.results.HCResults
         """
+        test_name = '{0}-missing-node-test'.format(ArakoonHealthCheck.MODULE)
 
-        arakoon_clusters, missing_nodes = ArakoonHealthCheck.fetch_clusters(logger)
+        fetched_clusters = ArakoonHealthCheck.fetch_clusters(result_handler)
+        missing_nodes = fetched_clusters['missing']
         if len([nodes for nodes in missing_nodes.itervalues() if len(nodes) != 0]) != 0:
```

Comment:
Same comment as last time, this makes no sense
This line is identical as line 341